src/user_crate/mod.rs

```diff
@@ -0,0 +1,333 @@
+                };
+                use ::pgx::{*, pg_sys};
+                struct PostAlloc;
+                #[global_allocator]
+                static PALLOC: PostAlloc = PostAlloc;
+                unsafe impl ::core::alloc::GlobalAlloc for PostAlloc {
+                    unsafe fn alloc(&self, layout: ::core::alloc::Layout) -> *mut u8 {
+                        ::pgx::pg_sys::palloc(layout.size()).cast()
+                    }
+                    unsafe fn dealloc(&self, ptr: *mut u8, _layout: ::core::alloc::Layout) {
+                        ::pgx::pg_sys::pfree(ptr.cast());
+                    }
+                    unsafe fn realloc(&self, ptr: *mut u8, _layout: Layout, new_size: usize) -> *mut u8 {
+                        ::pgx::pg_sys::repalloc(ptr.cast(), new_size).cast()
+                    }
+                }
+                #[pg_extern]
+                fn #symbol_ident(arg0: &str) -> Option<String> {
+                    Some(arg0.to_string())
+                }
+            };
+            assert_eq!(
+                generated_lib_rs,
+                fixture_lib_rs,
+                "Generated `lib.rs` differs from test (output formatted)\n\nGenerated:\n{}\nFixture:\n{}\n",
+                prettyplease::unparse(&generated_lib_rs),
+                prettyplease::unparse(&fixture_lib_rs)
+            );
+
+            let generated_cargo_toml = generated.cargo_toml()?;
+            let version_feature = format!("pgx/pg{}", pgx::pg_sys::get_pg_major_version_num());
+            let fixture_cargo_toml = toml! {
+                [package]
+                edition = "2021"
+                name = crate_name
+                version = "0.0.0"
+
+                [features]
+                default = [version_feature]
+
+                [lib]
+                crate-type = ["cdylib"]
+
+                [dependencies]
+                pgx = { version = "0.5.0-beta.0", features = [ "postgrestd" ] }
+
+                [profile.release]
+                debug-assertions = true
+                codegen-units = 1_usize
+                lto = "fat"
+                opt-level = 3_usize
+                panic = "unwind"
+            };
+            assert_eq!(
+                generated_cargo_toml,
+                *fixture_cargo_toml.as_table().unwrap(),
+                "Generated `Cargo.toml` differs from test (output formatted)\n\nGenerated:\n{}\nFixture:\n{}\n",
+                toml::to_string(&generated_cargo_toml)?,
+                toml::to_string(&fixture_cargo_toml)?,
+            );
+
+            let parent_dir = tempdir::TempDir::new("plrust-generated-crate-function-workflow")
+                .wrap_err("Creating temp dir")?;
+            let provisioned = generated.provision(parent_dir.path())?;
+
+            let (built, _output) =
+                provisioned.build(parent_dir.path(), pg_config, Some(target_dir.as_path()))?;
+
+            let _shared_object = built.shared_object();
+            // TODO: Assert
```

Comment:
Hm? What is this TODO waiting on, exactly? Assert what, the fcinfo condition?

Reply:
Uhhhh rereading it, I guess I wanted to assert a given symbol existed.